apisix/core/config_etcd.lua

```diff
@@ -545,7 +558,34 @@ local function _automatic_fetch(premature, self)
 
             local ok, err = sync_data(self)
             if err then
-                if err ~= "timeout" and err ~= "Key not found"
+                if string.find(err, "connection refused")
+                    or string.find(err, "Service Unavailable") then
+                    log.warn(err, ", ", tostring(self))
+                    local etcd_cli, err = get_etcd(true)
+                    if not etcd_cli then
+                        error("all etcd endpoints are unhealthy: " .. err)
+                    end
+                    self.etcd_cli = etcd_cli
+                    log.warn("changed to new etcd endpoint")
+                elseif string.find(err, "has no healthy etcd endpoint available") then
+                    log.warn(err, ", ", tostring(self))
+                    local reconnected = false
+                    while err and not reconnected do
+                        local backoff_duration, backoff_factor, backoff_step = 1, 2, 10
+                        for _ = 0, backoff_step, 1 do
+                            ngx_sleep(backoff_duration)
+                            _, err = sync_data(self)
+                            if not string.find(err, "has no healthy etcd endpoint available") then
```

Comment:
The `err` can be nil which will cause `find` to throw an error

Reply:
Fixed. Thanks!